tests/test_rollouts.py

```diff
@@ -82,6 +83,70 @@ async def test_rollout(training: bool) -> None:
     assert all(v == 2 for v in callback.fn_invocations.values())
 
 
+@pytest.mark.vcr
+@pytest.mark.parametrize("fallback", [True, False])
+@pytest.mark.asyncio
+async def test_fallbacks_working(fallback: bool) -> None:
+    AGENT_MODEL_LIST = [
+        {
+            "model_name": "openai/gpt-4o-mini",
+            "litellm_params": {"model": "openai/gpt-4o-mini", "api_key": "abc123"},
+        },
+        {
+            "model_name": "openai/gpt-4o",
+            "litellm_params": {
+                "model": "openai/gpt-4o",
+            },
+        },
+    ]
+    AGENT_ROUTER_KWARGS: dict[str, bool | list[dict[str, list[str]]]] = {
+        "set_verbose": True,
+    }
+    if fallback:
+        AGENT_ROUTER_KWARGS["fallbacks"] = [
+            {
+                "openai/gpt-4o-mini": [
+                    "openai/gpt-4o",
+                ]
+            }
+        ]
+
+    AGENT_CONFIG = {
+        "llm_model": {
+            "name": "openai/gpt-4o-mini",
+            "config": {
+                "model_list": AGENT_MODEL_LIST,
+                "router_kwargs": AGENT_ROUTER_KWARGS,
+            },
+        }
+    }
+    if fallback:
+        AGENT_CONFIG["llm_model"]["config"]["fallbacks"] = [  # type: ignore[index]
```

Comment:
We should assert the config is not mutated after the `SimpleAgent` is constructed